### PR TITLE
Fix bulk CSV

### PR DESCRIPTION
**Pull request by icreated**
* Refactoring / simplification bulk integration
* Fix CSV bulk uploading / creation graph
* Some twingraph endpoints better names

### CODE REVIEW

File: twingraph/src/main/kotlin/com/cosmotech/twingraph/api/TwingraphServiceImpl.kt [hunk 1]
```diff
@@ -74,11 +71,13 @@ class TwingraphServiceImpl(
     if (findAllTwingraphs(organizationId).contains(graphId))
         throw CsmServerException("There is already a graph with the id : $graphId")
 
+    val version = "1"
```
Comment (jreynard-code) [inline twingraph/src/main/kotlin/com/cosmotech/twingraph/api/TwingraphServiceImpl.kt:74]: Could be interesting to make it a `const` named `INITIAL_VERSION` for further usage :thinking: 
Maybe the same thing can be done for `DEFAULT_GRAPH_ROTATION` set to 3

**Review by lalepee — APPROVED** [review]

File: twingraph/src/main/kotlin/com/cosmotech/twingraph/api/TwingraphServiceImpl.kt [hunk 2]
```diff
@@ -334,9 +292,8 @@ class TwingraphServiceImpl(
       actionLambda: (Map<String, String>) -> Unit
   ): Map<String, String> {
     var map = mapOf<String, String>()
-    inputStream.bufferedReader(UTF_8).use { reader ->
-      val csvFormat: CSVFormat =
-          CSVFormat.DEFAULT.builder().setHeader().setSkipHeaderRecord(false).setTrim(true).build()
+    inputStream.bufferedReader().use { reader ->
```
Comment (jreynard-code) [inline twingraph/src/main/kotlin/com/cosmotech/twingraph/api/TwingraphServiceImpl.kt:295]: For information purpose, can you tell us why:
- UTF-8 is no longer used
- setSkipHeaderRecord is no longer used
- setTrim is no longer used

File: twingraph/src/main/kotlin/com/cosmotech/twingraph/bulk/PropertyType.kt [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  when (this) {
+    is Int -> type = PropertyType.BI_LONG
+    is Double -> type = PropertyType.BI_DOUBLE
+  }
+
+  (this as String).also {
+    if (it == "") type = PropertyType.BI_NULL
+    if (it.lowercase() == "true" || it.lowercase() == "false") type = PropertyType.BI_BOOL
+    if (it.startsWith('[') and it.endsWith(']')) type = PropertyType.BI_ARRAY
+  }
```
Comment (jreynard-code) [inline twingraph/src/main/kotlin/com/cosmotech/twingraph/bulk/PropertyType.kt:29]: Can be done in `when` block or I miss something? :thinking:

File: twingraph/src/main/kotlin/com/cosmotech/twingraph/bulk/model/Node.kt [hunk 4]
```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+package com.cosmotech.twingraph.bulk.model
+
+import com.cosmotech.twingraph.bulk.getNodeBinaryBlobFormat
+
+class Node(override val properties: Map<String, String>) : AbstractEntity() {
+
+  val id: String = (properties["id"] ?: properties.entries.iterator().next().value)
```
Comment (jreynard-code) [inline twingraph/src/main/kotlin/com/cosmotech/twingraph/bulk/model/Node.kt:9]: `(properties["id"] ?: properties.entries.iterator().next().value)` in fine you kept this?

File: twingraph/src/main/kotlin/com/cosmotech/twingraph/bulk/model/TypeEntity.kt [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+class TypeEntity(val type: String, private val header: List<String>) {
+
+  private val headerBinaryBlobFormat = getHeaderBinaryBlobFormat(type, header)
+  var size: Int = headerBinaryBlobFormat.size
+  var binaryBlobFormat: List<ByteArray> = mutableListOf()
+
+  fun addEntity(vararg nodesArg: AbstractEntity) {
+    nodesArg.forEach {
+      if (it.properties.keys.any { s -> s !in header })
+      // TODO create specific exception
```
Comment (jreynard-code) [inline twingraph/src/main/kotlin/com/cosmotech/twingraph/bulk/model/TypeEntity.kt:17]: Remove `TODO` or add a specific exception if needed ;)

**Review by jreynard-code — CHANGES_REQUESTED** [review]
LGTM :)
Some questions/ remarks btw ;)

File: twingraph/src/main/kotlin/com/cosmotech/twingraph/api/TwingraphServiceImpl.kt (hunk 2 above)
Comment (icreated) [reply twingraph/src/main/kotlin/com/cosmotech/twingraph/api/TwingraphServiceImpl.kt:295]: I will keep only trim. Others options are by default

File: twingraph/src/main/kotlin/com/cosmotech/twingraph/bulk/PropertyType.kt [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    is String -> {
+      when (this) {
+        "" -> type = PropertyType.BI_NULL
+        "true", "false" -> type = PropertyType.BI_BOOL
+        else -> {
+          if (this.lowercase() == "null") type = PropertyType.BI_NULL
+          if (this.lowercase() == "true" || this.lowercase() == "false") type = PropertyType.BI_BOOL
+          if (this.startsWith('[') and this.endsWith(']')) type = PropertyType.BI_ARRAY
+        }
+      }
```
Comment (jreynard-code) [inline twingraph/src/main/kotlin/com/cosmotech/twingraph/bulk/PropertyType.kt:32]: Could this be replaced by :

 ``` 
 when (this.lowercase()) {
        "" -> type = PropertyType.BI_NULL
        "true", "false" -> type = PropertyType.BI_BOOL
        else -> {
            if (this.startsWith('[') and this.endsWith(']')) type = PropertyType.BI_ARRAY
        }
      }
 ```

**Review by jreynard-code — CHANGES_REQUESTED** [review]
LGTM :)
Some syntactic sugar :)

**Review by jreynard-code — APPROVED** [review]
LGTM

**Pull request merged**